hazelcast/src/test/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceSlowTest.java

```diff
@@ -68,14 +68,19 @@ public void schedule_withLongSleepingCallable_blockingOnGet() throws Exception {
         IScheduledFuture<Double> future = executorService.schedule(
                 new ICountdownLatchCallableTask(initCountLatch.getName(), waitCountLatch.getName(), doneCountLatch.getName()), delay, SECONDS);
 
-        assertOpenEventually(initCountLatch);
-
         int sleepPeriod = 10000;
         long start = System.currentTimeMillis();
-        new Thread(() -> {
+        Thread thread = new Thread(() -> {
+            //This is to make sure that the executorService has scheduled the task
+            //and waiting for waitCountLatch
+            assertOpenEventually(initCountLatch);
```

Comment:
I didn't get this fix. When you look at the old state of the test, to start this thread, we have to see that the `initCountLatch` is open. And, it is only open when the executor service calls countdown on it. So, the executor service must have started before we call `thread.start`

Reply:
You are right. My fix is not correct.  I don't think the problem is about the threads. Probably it is about using different clocks for sleeping and measuring elapsed time. For sleeping System.nanoTime() is used but for measuring elapsed time System.currentTimeMillis(); is used.  It would be better to use System.nanoTime for everything. Do you want me to implement a fix like that ? Or do you want to assign the issue to somebody else and close the pull request ?

Reply:
I also believe that the issue is related to measurements on time. I don't think we have a flaw in the test logic or in the implementation. 

I think we can do the fix in this PR as you suggested.

Reply:
I agree. The actual fix is `thread.join` call I think.

Reply:
Not necessarily. I mean, yeah we could wait for the thread termination here, but normally, the `double result = future.get();` call will return after `waitCountLatch` is counted down. So, we were effectively waiting for the thread termination. Calling join or not will not help if the time reported by `System.currentTimeMillis()` is not inline with the amount of time we sleep with the help of `System.nanoTime()`

Reply:
I have changed the test to use System.nanoTime() to check for slept time